src/cmd/vendor.rs

```diff
@@ -0,0 +1,590 @@
+// Copyright (c) 2022 ETH Zurich
+// Michael Rogenmoser <michaero@iis.ee.ethz.ch>
+// Nils Wistoff <nwistoff@iis.ee.ethz.ch>
+
+//! The `vendor` subcommand.
+
+use crate::config::PrefixPaths;
+use crate::futures::TryFutureExt;
+use clap::{AppSettings, Arg, ArgMatches, Command};
+use futures::future::{self};
+use tokio::runtime::Runtime;
+
+use crate::config;
+use crate::error::*;
+use crate::git::Git;
+use crate::sess::{DependencySource, Session};
+use glob::Pattern;
+use std::path::Path;
+use std::path::PathBuf;
+use tempdir::TempDir;
+
+/// A patch linkage
+#[derive(Clone)]
+pub struct PatchLink {
+    /// directory
+    pub patch_dir: Option<PathBuf>,
+    /// prefix for upstream
+    pub from_prefix: PathBuf,
+    /// prefix for local
+    pub to_prefix: PathBuf,
+}
+
+/// Assemble the `vendor` subcommand.
+pub fn new<'a>() -> Command<'a> {
+    Command::new("vendor")
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .about("Copy source code from upstream external repositories into this repository. Functions similar to the lowrisc vendor.py script. Type bender vendor <SUBCOMMAND> --help for more information about the subcommands.")
+        .subcommand(Command::new("diff")
+            .about("Display a diff of the local tree and the upstream tree with patches applied.")
+        )
+        .subcommand(Command::new("init")
+            .about("(Re-)initialize the external dependencies. Copies the upstream files into the target directories and applies existing patches.")
+            .arg(
+                Arg::new("no_patch")
+                    .short('n')
+                    .long("no_patch")
+                    .help("Do not apply patches when initializing dependencies"),
+            )
+        )
+        .subcommand(Command::new("patch")
```

Comment:
Would it make sense to also add an optional argument `-m <msg>` to directly pass the message?